Flush occasionally stuck when SaveBinlogPath fails

<!-- Please state your issue using the following template and, most importantly, in English. -->

#### Steps/Code to reproduce:
Insert and Flush repeatably

#### Expected result:
Flush done without error

#### Actual results:
Occasionally flush will hang when SaveBinlogPath fails


#### Environment:
- Milvus version(e.g. v2.0.0-RC2 or 8b23a93): a7a6ff1
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

#### Additional context:
In datacoord log:
```
[2021/09/29 12:39:08.878 +00:00] [ERROR] [services.go:294] ["save binlog and checkpoints failed"] [segmentID=428059954682003458] [error="etcdserver: request timed out"] [stack="github.com/milvus-io/milvus/internal/datacoord.(*Server).SaveBinlogPaths\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/services.go:294\ngithub.com/milvus-io/milvus/internal/distributed/datacoord.(*Server).SaveBinlogPaths\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/service.go:234\ngithub.com/milvus-io/milvus/internal/proto/datapb._DataCoord_SaveBinlogPaths_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:2939\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/datapb._DataCoord_SaveBinlogPaths_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:2941\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]

```

See also
#7397
#8058

--- a/internal/datanode/flow_graph_insert_buffer_node.go
+++ b/internal/datanode/flow_graph_insert_buffer_node.go
@@ -261,7 +261,7 @@ func (ibNode *insertBufferNode) Operate(in []Msg) []Msg {
 		}
 	}
 
-	// Manul Flush
+	// Manual Flush
 	select {
 	case fmsg := <-ibNode.flushChan:
 		currentSegID := fmsg.segmentID
@@ -287,7 +287,7 @@ func (ibNode *insertBufferNode) Operate(in []Msg) []Msg {
 			finishCh := make(chan segmentFlushUnit, 1)
 
 			// Since buffer is not empty, so there must be data for key currentSegID
-			bd, _ := ibNode.insertBuffer.LoadAndDelete(currentSegID)
+			bd, _ := ibNode.insertBuffer.Load(currentSegID)
 
 			ibNode.flushMap.Store(currentSegID, bd.(*BufferData).buffer)
 			clearFn := func() {
@@ -324,9 +324,11 @@ func (ibNode *insertBufferNode) Operate(in []Msg) []Msg {
 					log.Debug("Data service save binlog path failed", zap.Error(err))
 				} else {
 					ibNode.replica.segmentFlushed(fu.segID)
-					ibNode.flushingSegCache.Remove(fu.segID)
+					ibNode.insertBuffer.Delete(fu.segID)
 				}
 			}
+			//always remove from flushing seg cache
+			ibNode.flushingSegCache.Remove(fu.segID)
 		}
 
 	default:

